pydantic/config.py

```diff
@@ -1014,6 +1014,23 @@ class Model(BaseModel):
         as the performance difference is minimal if repeated strings are rare.
     """
 
+    experimental_fast_build: bool
+    """Build model schemas faster by skipping the logic associated with storing the parent namespace on the model.
+
+    Defaults to `False` to maintain backwards compatibility, for now. The ideal case is that this change or it's successor
+    can become the default. Thus, if you experience issues with this setting that have not already been documented,
+    please open an issue on GitHub.
+
+    !!! warning
+        If this setting is set to `True`, model schemas for models that involve edge case uses of namespaces, such as
```

Comment:
Could we add unit tests showing which cases have issues when `experimental_fast_build=True`? So asserting the exception where they are given in the problem cases. Then its easier to understand what is the impact of using it. Docs could link to the unit test file showing the issues even.

Reply:
Great idea.

Reply:
Also would be nice to run some tests with `experimental_fast_build=True` where its expected to work.

Reply:
👍 tests added, thanks for the suggestion!